core/jobs/batch_jobs/skill_migration_jobs.py

```diff
@@ -0,0 +1,325 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs that are run by CRON scheduler."""
+
+from __future__ import annotations
+
+import logging
+
+from core import feconf
+from core.domain import caching_services
+from core.domain import skill_domain
+from core.domain import skill_fetchers
+from core.domain import skill_services
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+import result
+
+from typing import Iterable, Sequence, Tuple
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+    from mypy_imports import datastore_services
+    from mypy_imports import skill_models
+
+(base_models, skill_models,) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.skill
+])
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MigrateSkillJob(base_jobs.JobBase):
+    """Job that migrates skill models."""
+
+    @staticmethod
+    def _migrate_skill(
+        skill_id: str, skill_model: skill_models.SkillModel
+    ) -> result.Result[skill_domain.Skill, Exception]:
+        """Migrates skill and transform skill model into skill object.
+
+        Args:
+            skill_id: str. The id of the skill.
+            skill_model: SkillModel. The skill model to migrate.
+
+        Returns:
+            Result(Skill,Exception). Skill object when the migration
```

Comment:
Typically, in a tuple, add a space after each comma. Ditto below/elsewhere.

Also I am slightly confused about the typing here. Where does skill_id (which I see in the return values) come into it?

Reply:
Right, changed.